server/data/projects.js

```diff
@@ -93,9 +94,32 @@ const createProject = async (projectObjParam, user) => {
 	return createdProject;
 };
 
+const getSavedProjects = async (usernameParam, ownerParam) => {
+	let savedProjects;
+	const userName = isValidUsername(usernameParam);
+	const loggedinId = isValidObjectId(ownerParam);
+	const user = await getUserByUsername(userName);
+	const userId = user._id.toString();
+	if (userId === loggedinId) {
+		const projectsCollection = await projects();
+		savedProjects = await projectsCollection
+			.find({
+				savedBy: userId,
+			})
+			.toArray();
+		if (savedProjects.length === 0)
+			throw notFoundErr('You have not saved any project!');
```

Comment:
we can return an empty array instead like we do for list project, because we need to display this data in the frontend... 404 will show a failed network request

Reply:
Done.